iroh-docs/src/sync.rs

```diff
@@ -108,17 +108,34 @@ pub struct SyncOutcome {
     pub num_sent: usize,
 }
 
+fn get_as_ptr<T>(value: &T) -> Option<usize> {
+    use std::mem;
+    if mem::size_of::<T>() == std::mem::size_of::<usize>()
+        && mem::align_of::<T>() == mem::align_of::<usize>()
+    {
+        // Safe only if size and alignment requirements are met
+        unsafe { Some(mem::transmute_copy(value)) }
```

Comment:
Is this really the simplest way to check for pointer equality?

I'd much prefer https://github.com/smol-rs/async-channel/pull/98 to get merged soon.

Reply:
Yeah, me too. But what can you do...

This is fine as long as an async_channel remains an Arc<whatever>...